pydantic/_internal/_namespace_utils.py

```diff
@@ -0,0 +1,237 @@
+from __future__ import annotations
+
+import sys
+from collections.abc import Generator, Mapping
+from contextlib import contextmanager
+from functools import cached_property
+from typing import Any, Iterator, NamedTuple, TypeAlias, TypeVar
+
+GlobalsNamespace: TypeAlias = dict[str, Any]
```

Comment:
Do not hesitate if you have better ideas regarding naming. I first went with `GlobalsNamespace` and `LocalsNamespace`, but the latter was changed to `MappingNamespace` as I needed a type annotation for the fallback namespace, which can be the module globals as well (and this works because a module's `__dict__` is `dict[str, Any]`, assignable to `Mapping[str, Any]`).

Reply:
Happy with these as is for now - they're internal structures, so we can always bikeshed later. Resolving :).